### PR TITLE
ci: generate_gapic_bom as independent script

### PR DESCRIPTION
**Pull request by suztomo**

### CODE REVIEW

File: templates/generation/bom.pom.xml [hunk 1]
```diff
@@ -18,6 +18,17 @@ BOM_ARTIFACT_LIST
...
+    <snapshotRepository>
+      <id>sonatype-nexus-snapshots</id>
+      <url>https://google.oss.sonatype.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+    <repository>
+      <id>sonatype-nexus-staging</id>
+      <url>https://google.oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+    </repository>
+  </distributionManagement>
+
```
Comment (suztomo) [inline templates/generation/bom.pom.xml:31]: Added https://github.com/googleapis/google-cloud-java/pull/8180

File: bootstrap.sh [hunk 2]
```diff
@@ -61,33 +61,53 @@ do
   git remote add ${service} ../${service}
   git config --add secrets.allowed "dest.*src"
   git fetch ${service} #--tags
-  EDITOR=true git merge --allow-unrelated-histories ${service}/main
+  EDITOR=true git merge --quiet --allow-unrelated-histories ${service}/main
```
Comment (suztomo) [inline bootstrap.sh:64]: Adding "quiet" option. The output was too verbose.

File: bootstrap.sh [hunk 3]
```diff
@@ -114,8 +134,8 @@ for bom_directory in $(find . -name 'google-*-bom' | sort); do
       # increment the third digit of the version and overwrite it.
       snapshot_version=$(echo ${module_released_version} |  awk -F'.' '{print $1"."$2"."$3+1}' |  sed s/[.]$//)
 
-      mvn -f ${pom_directory} -U versions:set -DnewVersion=${snapshot_version}
-      mvn -f ${bom_directory} -U versions:set -DnewVersion=${snapshot_version}
+      mvn -B -ntp -f ${pom_directory} -U versions:set -DnewVersion=${snapshot_version}
+      mvn -B -ntp -f ${bom_directory} -U versions:set -DnewVersion=${snapshot_version}
```
Comment (suztomo) [inline bootstrap.sh:138]: Adding "quiet" option. The output was too verbose.

File: bootstrap.sh [hunk 4]
```diff
@@ -124,8 +144,8 @@ for bom_directory in $(find . -name 'google-*-bom' | sort); do
     # increment the third digit of the version and overwrite it.
     snapshot_version=$(echo ${module_released_version} |  awk -F'.' '{print $1"."$2"."$3+1}' |  sed s/[.]$//)
 
-    mvn -f ${pom_directory} -U versions:set -DnewVersion=${snapshot_version}
-    mvn -f ${bom_directory} -U versions:set -DnewVersion=${snapshot_version}
+    mvn -B -ntp -f ${pom_directory} -U versions:set -DnewVersion=${snapshot_version}
+    mvn -B -ntp -f ${bom_directory} -U versions:set -DnewVersion=${snapshot_version}
```
Comment (suztomo) [inline bootstrap.sh:148]: Adding "quiet" option. The output was too verbose.

File: bootstrap.sh [hunk 5]
```diff
@@ -172,57 +192,20 @@ for bom_directory in $(find . -name 'google-*-bom' | sort); do
...
-
 done
 
 echo "}" >> .release-please-manifest.json
 
 awk -v "packagesList=$rp_config_line" '{gsub(/ALL_PACKAGES/,packagesList)}1' \
     ../../release_please_config_raw.json > release-please-config.json
 
-mkdir google-cloud-gapic-bom
-cp ../../gapic_bom_versions.txt google-cloud-gapic-bom/versions.txt
```
Comment (suztomo) [inline bootstrap.sh:207]: This moved to line 101.

File: bootstrap.sh [hunk 6]
```diff
@@ -172,57 +192,20 @@ for bom_directory in $(find . -name 'google-*-bom' | sort); do
...
+git commit -am 'feat: create release please configuration'
 
 # Confirm everything is fine so far
 # Need license-checks.xml to validate
 mvn -q -B -ntp validate
 
-# Add all template files
-git add --all
-git add -f .gitignore
-git commit -m 'chore: add template files'
```
Comment (suztomo) [inline bootstrap.sh:224]: This moved to line 86.

File: templates/generation/generate_coverage_aggregator.sh [hunk 7]
```diff
@@ -1,9 +1,11 @@
 #!/bin/bash
 
+set -e
+
 GENERATION_DIR=$(dirname -- "$0");
 
 # generate coverage report
-mkdir CoverageAggregator
+mkdir -p CoverageAggregator
```
Comment (suztomo) [inline templates/generation/generate_coverage_aggregator.sh:8]: This "-p" allows to run this script against existing directory.

**Review by ddixit14 — APPROVED** [review]

**Pull request merged**